### PR TITLE
Minimally light up Quaternion and Plane for Mono SIMD

### LINKED ISSUE #81483: [Mono] Intrinsify Quaternion and Plane with LLVM as backend
`Quaternion` and `Plane` need to be intrinsified on all Mono supported platforms.

From Tanner (https://github.com/dotnet/runtime/pull/81335/files#r1091335649)
"Quaternion and Plane should be intrinsified. I don't have enough context into Mono to know if Plane is problematic since it is Vector3, float. I imagine Quaternion just needs to be handled since it has an identical layout to Vector4. However, not all the same functions get lightup for Quaternion and it is non-obvious how to correctly filter things here."

### PR DESCRIPTION
**Pull request by tannergooding**
Fixes #81483

### CODE REVIEW

**Comment by tannergooding** [comment]
Talked with @fanyang-mono a bit more.

Going to split this into at least 3 total PRs:
* This one, which adds Quaternion/Plane giving parity with the existing Vector4 support
* One that ensures new intrinsics, such as `SN_Length`, are handled
* One that adds support for Vector2/Vector3, since those require more in depth work

File: src/mono/mono/mini/simd-intrinsics.c [hunk 1]
```diff
@@ -301,8 +301,9 @@ emit_simd_ins_for_binary_op (MonoCompile *cfg, MonoClass *klass, MonoMethodSigna
 				instc0 = OP_FADD;
 				break;
 			case SN_Divide:
-			case SN_op_Division:
-				if (strcmp ("Vector4", m_class_get_name (klass)) && strcmp ("Vector2", m_class_get_name (klass))) {
+			case SN_op_Division: {
+				const char *class_name = m_class_get_name (klass);
+				if (strcmp ("Vector2", class_name) && strcmp ("Vector4", class_name) && strcmp ("Quaternion", class_name) && strcmp ("Plane", class_name)) {
```
Comment (fanyang-mono) [inline src/mono/mono/mini/simd-intrinsics.c:306]: Does it work for Vector2? Probably should move all the Vector2 related change to your future Vecor2/3 PR.

**Review by fanyang-mono — APPROVED** [review]

**Comment by fanyang-mono** [comment]
Thanks for adding intrinsics support of `Quaternion` and `Plane` for Mono.

File: src/mono/mono/mini/simd-intrinsics.c (hunk 1 above)
Comment (tannergooding) [reply src/mono/mono/mini/simd-intrinsics.c:306]: This is a filter of "don't execute this code for these types" where "Vector2" was already part of the condition.

I opted to not remove it, even though it isn't actually going to be hit today, given that it was pre-existing.

Comment (fanyang-mono) [reply src/mono/mono/mini/simd-intrinsics.c:306]: Gotcha!

**Comment by fanyang-mono** [comment]
Failures on `runtime (Build browser-wasm windows Release Mono_DebuggerTests_chrome)` is not related to this PR. It is a know issue #81792

**Comment by tannergooding** [comment]
Improvements:
* https://github.com/dotnet/perf-autofiling-issues/issues/12951

**Pull request merged**